data/crac-file/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/threshold/AbsoluteFlowThresholdTest.java

```diff
@@ -22,71 +22,125 @@
 
 /**
  * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
+ * @author Baptiste Seguinot {@literal <baptiste.seguino at rte-france.com>}
  */
 public class AbsoluteFlowThresholdTest {
 
+    private static final double DOUBLE_TOL = 0.5;
+
     private AbsoluteFlowThreshold absoluteFlowThresholdAmps;
     private AbsoluteFlowThreshold absoluteFlowThresholdMW;
     private Cnec cnec1;
     private Cnec cnec2;
+    private Cnec cnec3;
     private Network networkWithoutLf;
     private Network networkWithtLf;
 
     @Before
     public void setUp() {
+        absoluteFlowThresholdAmps = new AbsoluteFlowThreshold(Unit.AMPERE, Side.RIGHT, Direction.BOTH, 500.0);
+        absoluteFlowThresholdMW = new AbsoluteFlowThreshold(Unit.MEGAWATT, Side.LEFT, Direction.BOTH, 1500.0);
+
+        cnec1 = new SimpleCnec("cnec1", "cnec1", new NetworkElement("FRANCE_BELGIUM_1", "FRANCE_BELGIUM_1"),
+                absoluteFlowThresholdAmps, new SimpleState(Optional.empty(), new Instant("initial", 0)));
+
+        cnec2 = new SimpleCnec("cnec2", "cnec2", new NetworkElement("FRANCE_BELGIUM_1", "FRANCE_BELGIUM_2"),
+                absoluteFlowThresholdMW, new SimpleState(Optional.empty(), new Instant("initial", 0)));
 
```

Comment:
I would create an abstract class "AbstractThresholdTest" with a setUp method defining cnec1, cnec2, cnec3, networkWithoutLf and networkWithLf using Mockito.mock
Then in the concrete class test, you create the concrete objects (AbsoluteFlowThreshold or RelativeFlowThreshold) and you mock the behaviour of the mocked objects in the setup method.
It is not compulsory but I think it makes the unit tests more readable.

Reply:
good idea